src/browser/api_protocol/api_handlers/system.js

```diff
@@ -335,18 +336,28 @@ function SystemApiHandler() {
 
     function downloadAsset(identity, message, ack, errorAck) {
         let dataAck = _.clone(successAck);
-        System.downloadAsset(identity, message.payload, (err, dlId) => {
+        System.downloadAsset(identity, message.payload, (err) => {
             if (!err) {
-                dataAck.data = {
-                    dlId
-                };
                 ack(dataAck);
             } else {
                 errorAck(err);
             }
         });
     }
 
+    function downloadRuntime(identity, message, ack, nack) {
+        const { payload } = message;
+        const dataAck = Object.assign({}, successAck);
```

Comment:
that's a good point, I can use clone here and setup a task to move those to assign.